### ISSUE TITLE
Paddle取出最大特征值对应的特征向量

### ISSUE THREAD

**Issue by maye1998** [PaddlePaddle/Paddle#37632]
对于一个三维矩阵（batch_size, n, n）进行特征值分解

Tensorflow直接进行tf.linalg.eig本身就是排序好的，PyTorch的gather功能更强大一些，可以通过取最大拿到索引后得到。

但是paddle的gather没有那么强大，怎么能得到最大特征值对应的特征向量呢？

```
xx = np.random.randn(2, 3, 3)
XX = paddle.to_tensor(xx)
eig_value1, eig_vector1 = paddle.linalg.eig(XX)
indices = paddle.argmax(eig_value1.abs(), axis=-1, keepdim=True)
====================================
PyTorch可以根据indices和gather函数得到
但是paddle这个函数貌似是得不到
====================================
```

**Comment by danleifeng** [comment]
请问您使用gather的报错信息是这个op不支持complex型吗？

**Comment by Ligoml** [comment]
1. 取最大特征值对应的特征向量的问题：由于设计问题，paddle的确没有和tf对齐，这是符合预期的。在执行`paddle.linalg.eig`后，可以使用 `paddle.argmax` 取到最大特征值的索引，然后使用 `paddle.gather` 根据索引拿到特征向量
2. `paddle.gather `和 `torch.gather` 功能不一致的问题：这个问题不影响1的问题，此外paddle目前已有 `paddle.take_along_axis` 和 `torch.gather`功能等价，欢迎使用~
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/take_along_axis_cn.html#take-along-axis
